src/t8_cmesh/t8_cmesh_examples.c

```diff
@@ -2527,6 +2547,134 @@ t8_cmesh_new_full_hybrid (sc_MPI_Comm comm)
   return cmesh;
 }
 
+
+t8_cmesh_t
+t8_cmesh_new_pyramid_cake (sc_MPI_Comm comm, int num_of_pyra)
+{
+  
+  int                 current_pyra, current_face;
+  double             *vertices = T8_ALLOC (double, num_of_pyra * 5 * 3);
+  t8_cmesh_t          cmesh;
+  const double        degrees = 360. / num_of_pyra;
+  T8_ASSERT (num_of_pyra > 2);
+
+for (current_pyra= 0; current_pyra< num_of_pyra; current_pyra++) {
+    for (current_face = 0; current_face < 5; current_face++) {
+      /* Get the edges at the unit circle */
+      if (current_face == 4) {
+        vertices[current_pyra* 5 * 3 + current_face * 3] = 0;
+        vertices[current_pyra* 5 * 3 + current_face * 3 + 1] = 0;
+        vertices[current_pyra* 5 * 3 + current_face * 3 + 2] = 0;
+      }
+      else if (current_face == 1 || current_face == 3) {
+        vertices[current_pyra* 5 * 3 + current_face * 3] = cos (current_pyra* degrees * M_PI / 180);
+        vertices[current_pyra* 5 * 3 + current_face * 3 + 1] = sin (current_pyra* degrees * M_PI / 180);
+        vertices[current_pyra* 5 * 3 + current_face * 3 + 2] = (current_face == 3 ? 0.5 : -0.5);
+      }
+      else if (current_face == 0 || current_face == 2) {
+        vertices[current_pyra* 5 * 3 + current_face * 3] =
+          cos ((current_pyra* degrees + degrees) * M_PI / 180);
+        vertices[current_pyra* 5 * 3 + current_face * 3 + 1] =
+          sin ((current_pyra* degrees + degrees) * M_PI / 180);
+        vertices[current_pyra* 5 * 3 + current_face * 3 + 2] = (current_face == 2 ? 0.5 : -0.5);
+      }
+    }
+  }
+  t8_cmesh_init (&cmesh);
+  for (current_pyra = 0; current_pyra < num_of_pyra; current_pyra++) {
+    t8_cmesh_set_tree_class (cmesh, current_pyra, T8_ECLASS_PYRAMID);
+    t8_cmesh_set_join (cmesh, current_pyra, (current_pyra == (num_of_pyra - 1) ? 0 : current_pyra + 1), 0, 1, 0);
+    t8_cmesh_set_tree_vertices (cmesh, current_pyra, vertices + current_pyra * 15, 5);
+  }
+  t8_cmesh_commit (cmesh, comm);
+  T8_FREE (vertices);
+
+  return cmesh;
+}
+
+
+t8_cmesh_t
+t8_cmesh_new_long_brick_pyramid (sc_MPI_Comm comm, int num_cubes)
+{
+  double              vertices_coords[24] = {
+    0, 0, 0,
+    1, 0, 0,
+    0, 1, 0,
+    1, 1, 0,
+    0, 0, 1,
+    1, 0, 1,
+    0, 1, 1,
+    1, 1, 1
+  };
+  t8_locidx_t         vertices[5];
+  double              attr_vertices[15];
+  int                 current_cube, current_face;
+  t8_cmesh_t          cmesh;
+  T8_ASSERT (num_cubes > 0);
+  t8_cmesh_init (&cmesh);
+  for (current_cube = 0; current_cube < num_cubes; current_cube++) {
+    for (current_face = 0; current_face < 3; current_face++) {
```

Comment:
In this case, the first loop iterates over the amount of cubes and the second one over the amount of pyramids in each cube. If you visualize it in paraview, it will become better observable